diff --git a/.gitignore b/.gitignore
index 9a0fc86472..e5fca3c363 100644
--- a/.gitignore
+++ b/.gitignore
@@ -107,3 +107,15 @@ mindspore/lite/test/do_test/
 
 # lite opencl compile file
 *.cl.inc
+
+
+/KungFu
+/third_party/kungfu
+/mindspore/ccsrc/backend/kernel_compiler/cpu/kungfu
+/mindspore/ccsrc/backend/kernel_compiler/gpu/kungfu
+/mindspore/ops/operations/kungfu_comm_ops.py
+/mindspore/ccsrc/minddata/dataset/engine/datasetops/source/kungfu*
+/mindspore/ccsrc/minddata/dataset/engine/ir/datasetops/source/kungfu*
+/mindspore/nn/optim/kungfu.py
+/mindspore/train/callback/_kungfu.py
+
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a3b6d4c9b..2786492e4d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,6 +49,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/securec/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/flatbuffers/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/flatbuffers/include/flatbuffers)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/kungfu/include)
 
 include(${CMAKE_SOURCE_DIR}/cmake/dependency_utils.cmake)
 find_package(Python3 COMPONENTS Interpreter Development)
diff --git a/mindspore/ccsrc/CMakeLists.txt b/mindspore/ccsrc/CMakeLists.txt
index 4b6911416f..8bf692da8b 100644
--- a/mindspore/ccsrc/CMakeLists.txt
+++ b/mindspore/ccsrc/CMakeLists.txt
@@ -2,6 +2,7 @@
 include_directories(${CMAKE_SOURCE_DIR}/mindspore/core)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_BINARY_DIR})
+link_directories(${CMAKE_SOURCE_DIR}/third_party/kungfu/lib)
 
 if(ENABLE_CPU)
     if("${X86_64_SIMD}" STREQUAL "sse")
@@ -428,3 +429,12 @@ if(ENABLE_D)
 endif()
 
 add_subdirectory(cxx_api)
+
+if ($ENV{ENABLE_KUNGFU})
+    target_link_libraries(mindspore kungfu)
+    if (ENABLE_GPU)
+        target_link_libraries(mindspore kungfu_nccl)
+        target_link_libraries(mindspore mindspore::nccl)
+    endif()
+endif()
+
diff --git a/mindspore/nn/optim/__init__.py b/mindspore/nn/optim/__init__.py
index d7e0b05e1f..7c0455c4d2 100644
--- a/mindspore/nn/optim/__init__.py
+++ b/mindspore/nn/optim/__init__.py
@@ -30,6 +30,7 @@ from .proximal_ada_grad import ProximalAdagrad
 from .lazyadam import LazyAdam
 from .ada_grad import Adagrad
 from .thor import THOR
+from .kungfu import KungFuMomentum, KungFuSGD
 
 __all__ = ['Optimizer', 'Momentum', 'LARS', 'Adam', 'AdamWeightDecay', 'LazyAdam', 'AdamOffload',
-           'Lamb', 'SGD', 'FTRL', 'RMSProp', 'ProximalAdagrad', 'Adagrad', 'THOR']
+           'Lamb', 'SGD', 'FTRL', 'RMSProp', 'ProximalAdagrad', 'Adagrad', 'THOR', 'KungFuMomentum', 'KungFuSGD']
